readthedocs/api/v2/views/model_views.py

```diff
@@ -193,9 +192,9 @@ def sync_versions(self, request, **kwargs):  # noqa: D205
 
         # If the currently highest non-prerelease version is active, then make
         # the new latest version active as well.
-        old_highest_version = determine_stable_version(project.versions.all())
-        if old_highest_version is not None:
-            activate_new_stable = old_highest_version.active
+        current_stable = project.get_original_stable_version()
```

Comment:
Isn't this _checking_ for a new stable version (eg. the user has pushed a new higher version number), not getting it?

Reply:
That is done in the code below

https://github.com/readthedocs/readthedocs.org/blob/a387ac50463313b6c9eeaf06b210b0370b716d7c/readthedocs/api/v2/views/model_views.py#L242-L260

We just use this to get `activate_new_stable`